scripts/stop_rules.py

```
@@ -87,6 +87,160 @@ def load_reference(path: Path) -> dict:
+    """Re-optimize a trial's best-so-far trajectory geometries UNCONSTRAINED and return
+    the anytime running-min ABSOLUTE energy curve: (name, seed, d, [(n_calls, e_abs_eV)]).
+
+    Each improvement frame is a constrained best-so-far geometry; releasing it (tight
+    unconstrained opt) gives the true basin energy the reference is built on. Frames whose
+    unconstrained optimum changed connectivity (formed/broke a bond -> different species)
+    are dropped, mirroring the reference build and the solver's final-relax guard."""
+    import reference as ref  # same scripts/ dir; reuses the reference build machinery
+    from rdkit import Chem
+    from bouquet.setup import connectivity_changed
+
+    name = task["name"]; seed = task["seed"]
```

Ruff (0.15.21)

[error] Multiple statements on one line (semicolon)

(E702)